### DISCUSSION TITLE
Question regarding dependent behavior

### DISCUSSION THREAD

**Discussion by joep** [mongodb/mongoid#5619] [Q&A] [answered]
New to mongodb and mongoid and I basically copied an example from the docs (https://www.mongodb.com/docs/mongoid/current/reference/associations/#dependent-behavior). Code shown below.

However, when I delete a Band record the associated Album records do no seem to be deleted (which I verify in mongosh after executing the program) and instead appear to be orphaned.

Can someone please explain what I am not understanding? Thanks much!

    require 'mongoid'

    ENV["MONGOID_ENV"] = 'dbtesting'
    Mongoid.load!(File.join(File.dirname(__FILE__), 'mongoid.yml'))

    class Band
        include Mongoid::Document
        has_many :albums, dependent: :delete_all
    end
    
    class Album
        include Mongoid::Document
        belongs_to :band
    end

    Band.collection.drop()
    Album.collection.drop()
    p Band.count
    # result => 0
    p Album.count
    # result => 0
    r = Band.create()
    r.albums.create()
    r.albums.create()
    r.albums.create()
    r.albums.create()
    Band.first.delete # Will delete all associated albums.
    p Band.count
    # result => 0
    p Album.count
    # result => 4 ??????

**Comment by joep** [comment]
Sorry, I just realized this should be in the 'Q&A' category.

**Comment by comandeo** [comment, marked as answer]
Hi @joep,

The `delete` method only deletes the document itself. No callbacks are triggered, no associations are deleted. Contrary, the `destroy` method does what is expected. If you call `Band.first.destroy`, the albums will be deleted.

Thank you for pointing this out, I understand that the naming of the option and the documentation of the `delete` method can be improved.

**Reply by joep** [reply]
Hi @comandeo,

Thanks for the quick clarification. While the nomenclature is  a bit confusing, I think I might have dug deeper and resolved it on my own if not for the single line and comment in the documentation example:

`Band.first.delete # Will delete all associated albums.`

That comment just kept me focused on something I must be doing wrong.

**Reply by comandeo** [reply]
Oh, I did not notice that this line was also copied from the docs! This must be fixed, we raised a ticket for that - https://jira.mongodb.org/browse/MONGOID-5616

Thank you again for pointing this out!
